pkg/network/server.go

```diff
@@ -67,6 +67,8 @@ type (
 		connected *atomic.Bool
 
 		log *zap.Logger
+		// flag to check whether pong received in response.
+		pingSent bool
```

Comment:
This one certainly should not be server-wide, it's a peer relationship state flag.

Reply:
hm true. Will rework that. And I guess other points below should be reworked and moved to `peer` as well.